src/ReverseProxy/Service/RuntimeModel/Transforms/QueryParameterTransform.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.ReverseProxy.Service.RuntimeModel.Transforms
+{
+    internal abstract class QueryParameterTransform : RequestParametersTransform
+    {
+        private readonly QueryStringTransformMode _mode;
+        private readonly string _key;
+
+        public QueryParameterTransform(QueryStringTransformMode mode, string key)
+        {
+            _mode = mode;
+            _key = key;
+        }
+
+        public override void Apply(RequestParametersTransformContext context)
+        {
+            if (context == null)
+            {
+                throw new System.ArgumentNullException(nameof(context));
+            }
+
+            string value = GetValue(context);
+            if (!string.IsNullOrWhiteSpace(value))
```

Comment:
The WhiteSpace checks have some odd language issues, avoid them.
```suggestion
            if (!string.IsNullOrEmpty(value))
```